src/app/services/auth.service.ts

```diff
@@ -1,47 +1,83 @@
 import { Injectable } from '@angular/core';
 import { Router } from '@angular/router';
 import { AngularFireAuth } from 'angularfire2/auth';
-import * as firebase from 'firebase/app';
+/* import { AngularFirestore } from 'angularfire2/firestore';
+ */import * as firebase from 'firebase/app';
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/switchMap';
-import { StatsService } from './stats.service';
+
+/* interface User {
+    uid: string;
+    email: string;
+    photoURL?: string;
+    displayName?: string;
+} */
 
 @Injectable()
 export class AuthService {
+
+    user: Observable<firebase.User>;
+
     constructor(
-        private afAuth: AngularFireAuth,
-        private router: Router, public sts: StatsService) { }
-    get currentUser() {
-        return this.afAuth.auth.currentUser;
+        public afAuth: AngularFireAuth,
+/*         public afs: AngularFirestore,
+ */        public router: Router) {
+
+        // Get auth data, then get firestore user document || null
+        this.user = afAuth.authState;
+        /* this.user = afAuth.authState
+            .switchMap(user => {
+                if (user) {
+                    return this.afs.doc<User>(`users/${user.uid}`).valueChanges();
+                } else {
+                    return Observable.of(null);
+                }
+            }); */
     }
-    googleLogin() {
+    loginWithGoogle() {
         const provider = new firebase.auth.GoogleAuthProvider();
-        return this.oAuthLogin(provider)
+        this.afAuth.auth.signInWithPopup(provider)
             .then(value => {
-                console.log('Sucess', value),
-                    console.log('The given name is ' + value.additionalUserInfo.profile.given_name),
-                    this.router.navigateByUrl('/profile');
+                this.router.navigateByUrl('/game');
             })
             .catch(error => {
                 console.log('Something went wrong: ', error);
             });
     }
-    signInWithFacebook() {
-        this.deleteIfAnonymous();
-        return this.afAuth.auth.signInWithPopup(new firebase.auth.FacebookAuthProvider());
+    /* 
+    loginWithGoogle() {
+        const provider = new firebase.auth.GoogleAuthProvider();
+        return this.oAuthLogin(provider);
+    } *//* 
+    private oAuthLogin(provider) {
+        return this.afAuth.auth.signInWithPopup(provider)
+            .then(value => {
+                this.updateUserData(credential.user)
```

Comment:
remove or put in branch but that can't be in this Pr

Reply:
we removed all